test/unit/serializer/index.spec.ts

```diff
@@ -34,13 +56,58 @@ describe('index.ts', () => {
 					},
 				},
 			},
-			gzip: true,
 		});
+		const serializer = new Serializer(proto);
+		const write = await serializer.serialize(req);
+		const read = await serializer.deserialize(write);
+
+		expect(read).toMatchObject(req);
+	});
+
+	it('should work with fast json + gzip', async () => {
 		const req = {
 			bar: 'abc',
 		};
-		const write = await serializer.serialize(req, proto);
-		const read = await serializer.deserialize(write, proto);
+		const strategy = new JsonStrategy<typeof req>({
+			schema: {
+				title: 'Foo',
+				type: 'object',
+				properties: {
+					bar: {
+						type: 'string',
+					},
+				},
+			},
+		});
+		const serializer = new Serializer(strategy, new GzipStrategy());
+
+		const write = await serializer.serialize(req);
+		const read = await serializer.deserialize(write);
+
+		expect(read).toMatchObject(req);
+	});
+
+	it('should work with json', async () => {
+		const req = {
+			bar: 'abc',
+		};
+		const proto = new JsonStrategy<typeof req>();
+		const serializer = new Serializer(proto);
+		const write = await serializer.serialize(req);
+		const read = await serializer.deserialize(write);
+
+		expect(read).toMatchObject(req);
+	});
+
+	it('should work with json + gzip', async () => {
+		const req = {
+			bar: 'abc',
+		};
+		const strategy = new JsonStrategy<typeof req>();
+		const serializer = new Serializer(strategy, new GzipStrategy());
+
+		const write = await serializer.serialize(req);
+		const read = await serializer.deserialize(write);
 
```

Comment:
The test cases I wrote here can be used even if this suggestion is not accepted with minor adjustments